### ISSUE TITLE
Npm 7 workspace monorepo packaging

### ISSUE THREAD

**Issue by leemhenson** [FormidableLabs/serverless-jetpack#187]
Hi. I know there's the open PR regarding lerna monorepos using symlinks (https://github.com/FormidableLabs/serverless-jetpack/pull/142) but I'm currently working with a different arrangement which uses npm v7 workspaces, typescript project references, a flat directory structure and relative paths, e.g.:

```
/package.json
/node_modules   # <--- all dependencies get hoisted here, including the symlink to the /dist
/packages
  /common-a
    /package.json
    /dist/index.js
    /src/index.ts
  /common-b
    /package.json
    /dist/index.js
    /src/index.ts
  /function-1
    /package.json
    /dist/handler.js
    /src/hander.ts
```

Imports inside the handler, after compilation, look like `require("../../../common-a/dist/index")`. So, this is a path which escapes the confines of what serverless considers the `servicePath`. When jetpack produces the zip, it all looks great, except the handler paths are missing the necessary parent segment ("function-1") to make that relative path work:

```
common-a/dist/index.js
common-b/dist/index.js
dist/handler.js
```

I guess I'm looking for a way to say "prepend the function's file paths in the zip with 'function-1'". Is there any prior art in this sort of area?

Cheers!

**Comment by ryan-roemer** [comment]
What does your serverless.yml look like and which directory are you running the `serverless` commands from?

**Comment by leemhenson** [comment]
Hey Ryan, I found a reasonable solution:

1. put all my `serverless.yml`s alongside the root `node_modules`
2. name the `serverless.yml`s individually, e.g. `serverless.app-1.yml`
3. fully qualify the paths inside those config files, e.g. `handler: packages/app-1/dist/x/y/z/handler.handler`
4. use `sls -c serverless.app-1.yml <other stuff>`

Ideally i'd like to namespace the `.serverless` directories too, but that doesn't look possible right now. Either way, the serverless-jetpack is doing the biz exactly as I want now. And `serverless-offline` is happy with this arrangement too.

I don't need that other branch I made where I changed the trace algo to track nested node_modules + symlinks. That was a suboptimal monorepo layout anyway, as it turns out.
